packages/transactions/src/unsigned-transaction.ts

```diff
@@ -0,0 +1,17 @@
+import { ITransactionWithSignatures } from '.';
+import { BaseTransaction } from './types';
+
+export function getUnsignedTransaction<TTransaction extends BaseTransaction>(
+    transaction: TTransaction | (TTransaction & ITransactionWithSignatures)
+): TTransaction | Omit<TTransaction & ITransactionWithSignatures, keyof ITransactionWithSignatures> {
+    if ('signatures' in transaction) {
+        // The implication of the lifetime constraint changing is that any existing signatures are invalid.
+        const {
+            signatures: _, // eslint-disable-line @typescript-eslint/no-unused-vars
+            ...unsignedTransaction
+        } = transaction;
+        return unsignedTransaction;
```

Comment:
This one is for my own understanding: Through the act of making this a durable nonce transaction, we void any existing signatures? Bc they are only valid for a recent blockhash, right

Reply:
Basically the part of any transaction that needs to be signed _includes_ the blockhash/nonce as _content_. A change in the blockhash/nonce changes the transaction's content. Consequently, any existing signatures are no longer valid.